### PR TITLE
docs: Added Special Functions Page

### LINKED ISSUE #13036: docs: Add section or page for special functions
### Is your feature request related to a problem or challenge?

As mentioned in https://github.com/apache/datafusion/pull/12948#discussion_r1802076218 there are a few scalar functions that are not normal UDF's and cannot be documented in the standard manner as per https://github.com/apache/datafusion/issues/12740 Thus we need to find a way to add documentation for those functions to the user guide once the scalar function documentation is completely migrated over to the new code driven design.

### Describe the solution you'd like

A few idea:

- Hardcode a section into the generated scalar-functions.md for these functions
- Add a new page to the documentation that outlines 'special' functions

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jonathanc-n**
## Which issue does this PR close?

Closes #13036.

## What changes are included in this PR?
Added static special functions page for unnest

### CODE REVIEW

**Review by alamb — APPROVED** [review]
Thank you @jonathanc-n  -- we are so close.

I checked this out and built it locally. 
```shell
cd docs
./build.sh
open build/html/index.html
```

One thing I noticed was that there was a missing index entry, so I took the liberty of pushing a commit to fix that.

cc @Omega359

File: docs/source/user-guide/sql/special_functions.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+- **struct**: Object expression to unnest.
+  Can be a constant, column, or function, and any combination of object operators.
+
+#### Examples
+
+```
+> select * from foo;
++---------------------+
+| column1             |
```
Comment (comphead) [inline docs/source/user-guide/sql/special_functions.md:83]: I would rename the column name to struct_field or something. It required for me to look into the example twice to figure our the column1 is actually a structure. Renaming a column will make the example more intuitive

Comment (alamb) [reply docs/source/user-guide/sql/special_functions.md]: Good idea --done in 777f58de9

File: docs/source/user-guide/sql/special_functions.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+-->
+
+# Special Functions
+
+## Expansion Functiions
```
Comment (Omega359) [inline docs/source/user-guide/sql/special_functions.md:22]: tpyo

Comment (comphead) [inline docs/source/user-guide/sql/special_functions.md:22]:
```suggestion
## Expansion Functions
```

**Comment by alamb** [comment]
Thanks @jonathanc-n  and @comphead  for the careful review.

@Omega359  we are very close to finishing!

**Pull request merged**